src/os_posix/os_open.c

```diff
@@ -30,180 +84,77 @@ __open_directory(WT_SESSION_IMPL *session, char *path, int *fd)
  */
 int
 __wt_open(WT_SESSION_IMPL *session,
-    const char *name, bool ok_create, bool exclusive, int dio_type, WT_FH **fhp)
+    const char *name, int dio_type, u_int flags, WT_FH **fhp)
```

Comment:
Consistency, pretty much.  If you look at all the structures and function definitions with `flags` in them, they're nearly all `uint32_t` so to see one not like the others will make me pause and wonder why it is special/inconsistent, in the future.